File: app/src/main/java/com/example/shop/adapter/GoodsAdapter2.java
package com.example.shop.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shop.R;
import com.example.shop.entity.Goods;
import com.example.shop.util.T;

import java.util.List;

public class GoodsAdapter2 extends RecyclerView.Adapter<GoodsAdapter2.ViewHolder>{
    private List<Goods> mGoodsList;
    static class ViewHolder extends RecyclerView.ViewHolder{
        View mGoodsView;
        ImageView mIvGoods;
        ImageView mIvDelete;
        TextView mTvGoodsName;
        TextView mTvGoodsPrice;
        int gid;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            mGoodsView=itemView;
            mIvDelete=itemView.findViewById(R.id.iv_delete);
            mIvGoods=(ImageView) itemView.findViewById(R.id.iv_goods);
            mTvGoodsName=(TextView) itemView.findViewById(R.id.tv_goodsName);
            mTvGoodsPrice=(TextView) itemView.findViewById(R.id.tv_goodsPrice);
        }
    }
    public GoodsAdapter2(List<Goods> goodsList){
        mGoodsList=goodsList;
    }
    @NonNull
    @Override
    public GoodsAdapter2.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view= LayoutInflater.from(parent.getContext())
                .inflate(R.layout.goods2_item,parent,false);
        ViewHolder holder=new GoodsAdapter2.ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Goods goods =mGoodsList.get(position);
        holder.mIvGoods.setImageResource(goods.getImage());
        holder.mTvGoodsName.setText(goods.getTitle());
        holder.mTvGoodsPrice.setText(goods.getPrice());
        holder.gid=goods.getGid();
        holder.mIvDelete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                T.showShort(view.getContext(), "删除成功");
            }
        });
        holder.mGoodsView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                T.showShort(view.getContext(), "查看详情");
            }
        });
    }

    @Override
    public int getItemCount() {
        return mGoodsList.size();
    }
}

File: app/src/main/java/com/example/shop/adapter/GoodsAdapter2.kt
package com.example.shop.adapter


import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface
import com.example.shop.entity.Goods
import androidx.recyclerview.widget.RecyclerView
import android.widget.TextView
import com.example.shop.R
import android.view.ViewGroup
import android.view.LayoutInflater
import android.content.Intent
import android.net.Uri
import android.util.Log
import android.view.View
import android.widget.ImageView
import com.example.shop.MyApplication
import com.example.shop.activity.GoodsDetailActivity
import com.example.shop.activity.GoodsDetailActivity2
import com.example.shop.util.SetImageBitmapUtil
import com.example.shop.util.T
import kotlinx.coroutines.NonDisposableHandle.parent
import kotlinx.coroutines.currentCoroutineContext
import kotlinx.coroutines.runBlocking
import kotlin.random.Random

internal class GoodsAdapter2(private val mGoodsList: List<Goods>) : RecyclerView.Adapter<GoodsAdapter2.ViewHolder>() {
    private lateinit var mContext:Context
    internal class ViewHolder(var mGoodsView: View) : RecyclerView.ViewHolder(mGoodsView) {
        var mIvGoods: ImageView
        var mIvDelete: ImageView
        var mTvGoodsName: TextView
        var mTvGoodsPrice: TextView

        var gid = 0

        init {
            mIvDelete = itemView.findViewById(R.id.iv_delete)
            mIvGoods = itemView.findViewById<View>(R.id.iv_goods) as ImageView
            mTvGoodsName = itemView.findViewById<View>(R.id.tv_goodsName) as TextView
            mTvGoodsPrice = itemView.findViewById<View>(R.id.tv_goodsPrice) as TextView
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        mContext=parent.context
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.goods2_item, parent, false)

        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val goods = mGoodsList[position]
        //holder.mIvGoods.setImageResource(goods.image)
        holder.mIvGoods.setImageURI(Uri.parse(goods.image))
        //SetImageBitmapUtil.setBitmap(goods.image,holder.mIvGoods,mContext)
        holder.mTvGoodsName.text = goods.title
        holder.mTvGoodsPrice.text = goods.price
        holder.gid = goods.gid
        holder.mIvDelete.setOnClickListener { view ->
            showDeleteDialog(mContext,goods)

        }
        holder.mGoodsView.setOnClickListener { view ->
            val position = holder.adapterPosition
            val goods = mGoodsList[position]
            val intent = Intent()
            intent.putExtra("gid", goods.gid)
            intent.setClass(view.context, GoodsDetailActivity2::class.java)
            view.context.startActivity(intent)
            T.showShort(view.context, "????????????")
        }
    }

    override fun getItemCount(): Int {
        return mGoodsList.size
    }
    private fun showDeleteDialog(context: Context,goods:Goods) {
        val builder = AlertDialog.Builder(context)
        builder.setTitle("????????????")
        builder.setMessage("????????????????????????")

        val dialogClickListener = DialogInterface.OnClickListener { _, which ->
            when (which) {
                DialogInterface.BUTTON_POSITIVE -> {
                    runBlocking {
                        MyApplication.instance.goodsDao.deleteGoods(goods)
                    }
                    T.showShort(context, "????????????")
                }
                DialogInterface.BUTTON_NEGATIVE -> {
                    Log.e("TAG", "??????")
                }
            }
        }
        builder.setPositiveButton("??????", dialogClickListener)
        builder.setNegativeButton("??????", dialogClickListener)
        builder.create().show();
    }
}